canonical-web-and-design/maas-ui
# feat(ui): make machine delete and set zone forms works for all node types

## Done

- Refactored `SetZoneForm` to work with all node types
- Extracted `DeleteForm` from `FieldlessForm` and made it work across node types

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Note there is currently a bug where the forms won't work properly when using a newly added device. If you add a new device, refresh the page to get state in the right place. https://github.com/canonical-web-and-design/app-tribe/issues/587
- Go to the device list and check that you can set zone and delete selected devices
- Go to a device details page and check that you can set the zone there
- Check that deleting a device from the details view redirects you to the device list
- Check that deleting/setting zone for machines still works in the list and details views

## Fixes

Fixes canonical-web-and-design/app-tribe#584
